symbolic link trouble + PHPUnit unknown

| Question    | Answer
| ------------| ---------------
| Infection version | 0.13.6
| Test Framework version | PHPUnit 7.5.15
| PHP version | 7.3.5
| Platform | Windows 10 Pro (using ConEmu)
| Github Repo | https://github.com/ccwebdesign/infection-temp

When run under a symbolic link in a terminal, infection creates all of the mutants, but claims they're all uncovered by tests.

To reproduce:
- create a directory with some code, tests, and Infection
- create a symbolic link to that directory
- cd into the symbolic link and run tests then infection

Repository above contains samples and structure if you want it.

Infection was run from the test subdirectory with:
```
..\..\vendor\bin\infection --coverage=tmp -j=4
```

Separate issue is that the PHPUnit version is unknown, both in the symlink folder and the actual one. (Whether vendor/bin directory is or isn't in PATH.)

Left out the phpunit.junit.xml for brevity (and since it's probably not relevant).

<details>
 <summary>Output with issue</summary>
 
 ```
You are running Infection with Xdebug enabled.
     ____      ____          __  _
    /  _/___  / __/__  _____/ /_(_)___  ____
    / // __ \/ /_/ _ \/ ___/ __/ / __ \/ __ \
  _/ // / / / __/  __/ /__/ /_/ / /_/ / / / /
 /___/_/ /_/_/  \___/\___/\__/_/\____/_/ /_/

    0 [>---------------------------] < 1 secRunning initial test suite...

PHPUnit version: unknown

    1 [============================] < 1 secProcessing source code files:   0/585

Generate mutants...

Processing source code files: 585/585Creating mutated files and processes:    0/6999
Creating mutated files and processes: 6999/6999
...
6999 mutations were generated:
       0 mutants were killed
    6999 mutants were not covered by tests
       0 covered mutants were not detected
       0 errors were encountered
       0 time outs were encountered
 ```
</details>

Steps to reproduce:
```
git clone https://github.com/ccwebdesign/infection-temp
cd infection-temp/
composer install
cd lib/App/test/
../../vendor/bin/phpunit
../../vendor/bin/infection
```
The last command outputs:
```
$ ../../vendor/bin/infection 
You are running Infection with Xdebug enabled.
     ____      ____          __  _
    /  _/___  / __/__  _____/ /_(_)___  ____
    / // __ \/ /_/ _ \/ ___/ __/ / __ \/ __ \
  _/ // / / / __/  __/ /__/ /_/ / /_/ / / / /
 /___/_/ /_/_/  \___/\___/\__/_/\____/_/ /_/

Running initial test suite...

PHPUnit version: 7.5.15

    1 [============================] < 1 sec
                                                                                                                        
 [ERROR] Project tests must be in a passing state before running Infection.                                             
                                                                                                                        
         Infection runs the test suite in a RANDOM order. Make sure your tests do not have hidden dependencies.         
                                                                                                                        
         You can add these attributes to `phpunit.xml` to check it: <phpunit executionOrder="random"                    
         resolveDependencies="true" ...                                                                                 
                                                                                                                        
         If you don't want to let Infection run tests in a random order, set the `executionOrder` to some value, for    
         example <phpunit executionOrder="default"                                                                      
                                                                                                                        
         Check the executed command to identify the problem: '/tmp/infection-temp/lib/vendor/phpunit/phpunit/phpunit'   
         '--configuration'                                                                                              
         '/tmp/infection-temp/lib/App/test/tmp/infection/infection/phpunitConfiguration.initial.infection.xml'          
                                                                                                                        
         PHPUnit reported an exit code of 143.                                                                          
                                                                                                                        
         Refer to the PHPUnit's output below:                                                                           
                                                                                                                        
         STDERR:                                                                                                        
                                                                                                                        
         PHPUnit 7.5.15 by Sebastian Bergmann and contributors.                                                         
                                                                                                                        
         Runtime:       PHP 7.3.8-1 with Xdebug 2.7.2                                                                   
         Configuration:                                                                                                 
         /tmp/infection-temp/lib/App/test/tmp/infection/infection/phpunitConfiguration.initial.infection.xml            
         Random seed:   1568713992                                                                                      
                                                                                                                        
                                                                                                                        
```

--- a/src/TestFramework/PhpUnit/Config/Builder/InitialConfigBuilder.php
+++ b/src/TestFramework/PhpUnit/Config/Builder/InitialConfigBuilder.php
@@ -109,6 +109,7 @@ class InitialConfigBuilder implements ConfigBuilder
         $this->xmlConfigurationHelper->setStopOnFailure($xPath);
         $this->xmlConfigurationHelper->deactivateColours($xPath);
         $this->xmlConfigurationHelper->deactivateResultCaching($xPath);
+        $this->xmlConfigurationHelper->deactivateStderrRedirection($xPath);
         $this->xmlConfigurationHelper->removeExistingLoggers($dom, $xPath);
         $this->xmlConfigurationHelper->removeExistingPrinters($dom, $xPath);
 

--- a/src/TestFramework/PhpUnit/Config/Builder/MutationConfigBuilder.php
+++ b/src/TestFramework/PhpUnit/Config/Builder/MutationConfigBuilder.php
@@ -89,6 +89,7 @@ class MutationConfigBuilder extends ConfigBuilder
         $this->xmlConfigurationHelper->setStopOnFailure($xPath);
         $this->xmlConfigurationHelper->deactivateColours($xPath);
         $this->xmlConfigurationHelper->deactivateResultCaching($xPath);
+        $this->xmlConfigurationHelper->deactivateStderrRedirection($xPath);
         $this->xmlConfigurationHelper->removeExistingLoggers($dom, $xPath);
         $this->xmlConfigurationHelper->removeExistingPrinters($dom, $xPath);
         $this->xmlConfigurationHelper->removeDefaultTestSuite($dom, $xPath);

--- a/src/TestFramework/PhpUnit/Config/XmlConfigurationHelper.php
+++ b/src/TestFramework/PhpUnit/Config/XmlConfigurationHelper.php
@@ -85,14 +85,12 @@ final class XmlConfigurationHelper
 
     public function deactivateResultCaching(\DOMXPath $xPath): void
     {
-        $nodeList = $xPath->query('/phpunit/@cacheResult');
+        $this->setAttributeValue($xPath, 'cacheResult', 'false');
+    }
 
-        if ($nodeList->length) {
-            $nodeList[0]->nodeValue = 'false';
-        } else {
-            $node = $xPath->query('/phpunit')[0];
-            $node->setAttribute('cacheResult', 'false');
-        }
+    public function deactivateStderrRedirection(\DOMXPath $xPath): void
+    {
+        $this->setAttributeValue($xPath, 'stderr', 'false');
     }
 
     public function setStopOnFailure(\DOMXPath $xPath): void

--- a/tests/phpunit/TestFramework/PhpUnit/Config/XmlConfigurationHelperTest.php
+++ b/tests/phpunit/TestFramework/PhpUnit/Config/XmlConfigurationHelperTest.php
@@ -415,6 +415,71 @@ XML
         );
     }
 
+    public function test_it_sets_stderr_to_false_when_it_exists(): void
+    {
+        $dom = new \DOMDocument();
+
+        $dom->preserveWhiteSpace = false;
+        $dom->formatOutput = true;
+
+        $dom->loadXML(<<<XML
+<?xml version="1.0" encoding="UTF-8"?>
+<phpunit
+    stderr="true"
+    syntaxCheck="false"
+>
+</phpunit>
+XML
+        );
+
+        $xmlconfig = new XmlConfigurationHelper($this->getPathReplacer(), '');
+
+        $xmlconfig->deactivateStderrRedirection(new \DOMXPath($dom));
+
+        $this->assertXmlStringEqualsXmlString(<<<XML
+<?xml version="1.0" encoding="UTF-8"?>
+<phpunit
+    stderr="false"
+    syntaxCheck="false"
+>
+</phpunit>
+XML
+            , $dom->saveXML()
+        );
+    }
+
+    public function test_it_sets_stderr_to_false_when_it_does_not_exist(): void
+    {
+        $dom = new \DOMDocument();
+
+        $dom->preserveWhiteSpace = false;
+        $dom->formatOutput = true;
+
+        $dom->loadXML(<<<XML
+<?xml version="1.0" encoding="UTF-8"?>
+<phpunit
+    syntaxCheck="false"
+>
+</phpunit>
+XML
+        );
+
+        $xmlconfig = new XmlConfigurationHelper($this->getPathReplacer(), '');
+
+        $xmlconfig->deactivateStderrRedirection(new \DOMXPath($dom));
+
+        $this->assertXmlStringEqualsXmlString(<<<XML
+<?xml version="1.0" encoding="UTF-8"?>
+<phpunit
+    stderr="false"
+    syntaxCheck="false"
+>
+</phpunit>
+XML
+            , $dom->saveXML()
+        );
+    }
+
     public function test_it_removes_existing_printers(): void
     {
         $dom = new \DOMDocument();

